include/sys/sysmacros.h

```diff
@@ -0,0 +1,40 @@
+/****************************************************************************
+ * include/sys/sysmacros.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_SYS_SYSMACROS_H
+#define __INCLUDE_SYS_SYSMACROS_H
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define major(x) \
+        ((unsigned)((((x) >> 31 >> 1) & 0xfffff000) | \
+                    (((x) >> 8) & 0x00000fff)))
+#define minor(x) \
+        ((unsigned)((((x) >> 12) & 0xffffff00) | ((x) & 0x000000ff)))
+
+#define makedev(x,y) ( \
+        (((x) & 0xfffff000ull) << 32) | \
```

Comment:
Definitely depends on `HAVE_LONG_LONG`

Reply:
ULL to big, refer to :https://github.com/freebsd/freebsd-src/blob/833a452e9f082a7982a31c21f0da437dbbe0a39d/contrib/libarchive/libarchive/archive_pack_dev.c#L90 to change to uint32.